### DISCUSSION TITLE
check image quality for AR Tracking

### DISCUSSION THREAD

**Discussion by Arun1947** [sceneview/sceneform-android#338] [General]
How to check the quality of the image added for AR tracking?
Does it support it or not?

**Comment by ThomasGorisse** [comment]
Please make a simple quick search before asking here: [ARCore Check image quality](https://developers.google.com/ar/develop/augmented-images/arcoreimg#check_image_quality)

**Comment by Arun1947** [comment]
I mean, how to check this inside android app ?

**Comment by RGregat** [comment]
This is not SceneView/SceneForm related, but a quick look to the AugmentedImageDatabase class gives you a hint

```
com.google.ar.core.exceptions.ImageInsufficientQualityException – if the image quality is image is insufficient, e.g. if the image has not enough features.
IllegalArgumentException – if the bitmap is not in ARGB_888 format or the width in meters is less than or equal to zero
```
 
If you don't get an exception the image is usable.
